carrot_planner/src/carrot_planner.cpp

```diff
@@ -155,7 +149,8 @@ namespace carrot_planner {
 
     plan.push_back(start);
     geometry_msgs::PoseStamped new_goal = goal;
-    tf::Quaternion goal_quat = tf::createQuaternionFromYaw(target_yaw);
+    tf2::Quaternion goal_quat;
+    goal_quat.setEuler(target_yaw, 0, 0);
```

Comment:
This is one potential issue -- it looks like internally createQuaternionFromYaw() called setRPY(). In TF2 -- setEuler() is different from SetRPY() or SetEulerZYX() which internally calls SetRPY()